data_structures/linked-list/linked_list.py

```
@@ -41,66 +41,74 @@ def find(self, val):
+                        previous._next = new_node
+                        self._size += 1
+                previous = search
+                search = search._next
+        except (KeyError, ValueError):
+            raise Exception('Value not found!')
 
     def insert_after(self, val, newVal):
         """Insert a new node after the matching value."""
         search = self.head
-        while search:
-            if search.val == val:
-                reference = search._next
-                search._next = Node(newVal)
-                search._next._next = reference
-                self._size += 1
-                break
-            search = search._next
+        try:
+            while search:
+                if search.val == val:
+                    reference = search._next
+                    search._next = Node(newVal)
+                    search._next._next = reference
+                    self._size += 1
+                search = search._next
+        except (KeyError, ValueError):
+            raise Exception('Value not found!')
 
     def kthFromEnd(self, k):
         """Return value of node at k positions from end."""
         main = self.head
         reference = self.head 
         counter = 0
-        if type(k) is not int:
-            raise Exception('Invalid value! Please enter an integer')
-        if k == 0:
-            raise Exception('Invalid value! Please enter a number greater than 0.')
-        if(self.head is not None):
-            while(counter < k):
-                if(reference is None):
-                    return
+        try:
+            if type(k) is not int:
+                raise Exception('Invalid value! Please enter an integer')
+            if k == 0:
+                raise Exception('Invalid value! Please enter a number greater than 0.')
+            if(self.head is not None):
+                while(counter < k):
+                    if(reference is None):
+                        return
```

This means that the given `k` is outside the length of the list, thus this should raise an exception of some type.